OpenRA.Mods.Cnc/Traits/OnRails.cs

```diff
@@ -0,0 +1,211 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Activities;
+using OpenRA.Mods.Common;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Cnc.Traits
+{
+	public class OnRailsInfo : ITraitInfo, IPositionableInfo, IFacingInfo, IMoveInfo,
```

Comment:
How many other mods are likely to need a unit that has exactly two artwork facings and moves left-right bouncing off the edge of the map?

I think it would be better to call this `TDGunboat` or similar so that we keep `OnRails` available for a less specific use case.